library.c

```diff
@@ -1470,11 +1471,13 @@ PHP_REDIS_API int redis_sock_connect(RedisSock *redis_sock TSRMLS_DC)
         return -1;
     }
 
-    /* Attempt to set TCP_NODELAY if we're not using a unix socket. */
+    /* Attempt to set TCP_NODELAY/TCP_KEEPALIVE if we're not using a unix socket. */
     sock = (php_netstream_data_t*)redis_sock->stream->abstract;
     if (!usocket) {
         err = setsockopt(sock->socket, IPPROTO_TCP, TCP_NODELAY, (char *) &tcp_flag, sizeof(int));
         PHPREDIS_NOTUSED(err);
+        err = setsockopt(sock->socket, SOL_SOCKET, SO_KEEPALIVE, (char *) &redis_sock->tcp_keepalive, sizeof(int));
```

Comment:
`int setsockopt(int sockfd, int level, int optname, const void *optval, socklen_t optlen);`
Please change fourth argument type from `char *` to `const void *`